build/parseChangelog.c

```diff
@@ -130,8 +130,7 @@ static int dateToTimet(const char * datestr, time_t * secs, int * date_words)
 	SKIPNONSPACE(pe);
 	if (*pe != '\0')
 	   *pe++ = '\0';
-	if (((int)(pe-p) + 1) > 9 )
-	   goto exit;
+	tz_name = xmalloc((int)(pe-p)+1);
 	strncpy(tz_name, p, (int)(pe-p));
 	tz_name[(int)(pe-p)] = '\0';
```

Comment:
The same pe-p calculation is done several times here with cases and all. Use a helper variable in a case like this.

Reply:
Oh and the casts are unnecessary here anyway. Never cast unless you really, really have to, and even then there's usually a better way.